### PR TITLE
[HEXAGON] Disable automatic unrolling in LLVM

### PR DESCRIPTION
**Pull request by joshherr-quic**
Automatic unrolling in LLVM was causing significant issues with compile time and binary size of a bert int8 model. This patch disables unrolling in LLVM for the hexagon backend. While not a permanent solution, this should unblock some development efforts on this path.

cc @mehrdadh

### CODE REVIEW

**Comment by mehrdadh** [comment]
@joshherr-quic please update the PR description :)

**Comment by cconvey** [comment]
I see that this is failing CI on Hexagon with this error ([link](https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-12639/1/pipeline/433/#step-1524-log-205)): 

```
[2022-08-29T22:19:07.000Z] tests/python/contrib/test_hexagon/test_2d_physical_buffers.py::TestElementWise::test_build[nhwc-nhwc-int8-16-32-32x32-nhwc-global-hexagon] LLVM ERROR: Cannot select: 0x3fbc580: v8i8 = HexagonISD::VASL 0x3e53240, Constant:i32<1>
[2022-08-29T22:19:07.000Z]   0x3e53240: v8i8,ch = load<(load (s64) from %ir.cgep31, align 64, !tbaa !97)> 0x3da4928, 0x3d14738, undef:i32
[2022-08-29T22:19:07.000Z]     0x3d14738: i32 = add 0x3d14390, Constant:i32<-128>
[2022-08-29T22:19:07.000Z]       0x3d14390: i32,ch = CopyFromReg 0x3da4928, Register:i32 %14
[2022-08-29T22:19:07.000Z]         0x3d14a78: i32 = Register %14
[2022-08-29T22:19:07.000Z]       0x3d13f80: i32 = Constant<-128>
[2022-08-29T22:19:07.000Z]     0x3f81c38: i32 = undef
[2022-08-29T22:19:07.000Z]   0x3d14120: i32 = Constant<1>
[2022-08-29T22:19:07.000Z] In function: default_function_compute_
[2022-08-29T22:19:07.000Z] Fatal Python error: Aborted
```

If CI is using LLVM build provided by Hexagon SDK 4.5.0.3 ([link](https://github.com/apache/tvm/blob/81b477aa4fc4c42f889fdec1a8064916f90bcf57/docker/Dockerfile.ci_hexagon#L20)), does this mean CI would need a newer version of the Hexagon SDK for this PR?

**Comment by supersat** [comment]
Is this crash solved by `https://github.com/llvm/llvm-project/commit/3c817574c2db9dbb003914dc40d58fe1dcfda855`? If so, this could be gated on `TVM_LLVM_VERSION >= 160`

**Comment by joshherr-quic** [comment]
> Contributor

**Comment by joshherr-quic** [comment]
Accidentally closed....

**Comment by joshherr-quic** [comment]
> Is this crash solved by `https://github.com/llvm/llvm-project/commit/3c817574c2db9dbb003914dc40d58fe1dcfda855`? If so, this could be gated on `TVM_LLVM_VERSION >= 160`

That's correct, I'll go ahead and gate it and push a followup commit.

File: src/target/llvm/codegen_hexagon.cc [hunk 1]
```diff
@@ -513,6 +513,13 @@ runtime::Module BuildHexagon(IRModule mod, Target target) {
                           {"-hexagon-small-data-threshold=0",
                            "-force-target-max-vector-interleave=1", "-hexagon-autohvx=1"});
 
+  // Disable unrolling (bug fix in LLVM 16 for int8 vectors)
+  // TODO: figure out why unroll is causing significant compile
+  // time and binary size issues for bert int8 model among others.
+#if TVM_LLVM_VERSION >= 160
+  llvm_options_vec.insert({"-unroll-threshold=0"})
+#endif
```
Comment (cconvey) [inline src/target/llvm/codegen_hexagon.cc:521]: Would it make sense to (in some manner) control this via cmake?

I'm just wondering if some developers will want to use _patched_ LLVM builds that _should_ make use of this option, but which don't cause `TVM_LLVM_VERSION >= 160` to be true.

I don't know either way, just figured I'd ask the question.

Comment (cconvey) [reply src/target/llvm/codegen_hexagon.cc]: That makes sense.  If someone really needs to work around this, they can also just comment-out that `#ifdef` temporarily.

**Review by cconvey — APPROVED** [review]
LGTM!

**Pull request closed without merging**